CONTRIBUTING_WITH_GIT2R.Rmd

```diff
@@ -0,0 +1,551 @@
+list.files()
+```
+
+```{r show_shadow, echo = FALSE}
+list.files(fork_dir)
+```
+
+
+We can move into the folder with `setwd()` and look at all the files in there.
+
+```{r go_clone, eval = FALSE}
+setwd("popgenInfo")
+list.files()
+```
+
+```{r go_shadow, echo = FALSE}
+fork_pgi <- paste0(fork_dir, "/popgenInfo")
+list.files(fork_pgi)
+```
+
+Now we have successfully cloned our repository into our computer using *git2r*. 
+Next, we will set up our credentials and keep our master branch up to date.
+
+## Setting up your clone
+
+To access the repository, all you have to do is open it by double-clicking on
+the "popgenInfo.Rproj" file. This will open Rstudio to this folder and set it as
+the working directory. After that, in your R console, type:
+
+```{r repository_show, eval = FALSE}
+library("git2r") # in case you're starting from here
+repo <- repository()
+repo
+```
+
+```{r repository_shadow, echo = FALSE}
+library("git2r") # in case you're starting from here
+repo <- repository(fork_pgi)
+repo
+```
+
+This tells R that you have a repository in this folder and that you are
+referring to it as `repo`. The output of `repo` is a summary of your repository:
+
+ - **Local** This is the branch (master) and the path where the copy of your repository is.
+ - **Remote** This shows you the branch and the url of the remote repository
+ - **Head** This is a summary of the last *commit* containing the date and a description.
+
+In order to be able to synch your clone with your fork, GitHub needs to know 
+that you are really who you say you are. This means that you need to associate
+your name, email, and a secret token with this repository. The first two items
+are easily done within R. You can use the function `config()` to set and confirm
+your name and email:
+
+```{r, config}
+config(repo, 
+       user.name = "Zhian Kamvar",
+       user.email = "kamvarz@science.oregonstate.edu")
+config(repo)
+```
+
+Next, we need to set our secret token. Known as a [**Personal Access Token
+(PAT)**](https://help.github.com/articles/creating-an-access-token-for-command-line-use/),
+this is a 40 character cryptographic token that is like a long and complicated
+password. They both allow access to your GitHub repository, but the difference
+between a PAT and a password is that a PAT can be stored as text on your
+computer (which you should NEVER do with passwords) because it is easy to
+generate and (most importantly) easy to remove. We can create a PAT from GitHub
+and then we will store this in a file called `.Renviron` that is read every time
+you restart your R session, allowing you to have this token whenever you need to
+push changes to your fork on GitHub. [The instructions to generate your PAT are
+here](https://help.github.com/articles/creating-an-access-token-for-command-line-use/).
+
+Here's an example of what a `.Renviron` file with a PAT will look like on the
+inside:
+
+```
+GITHUB_PAT="97cc6bf86c31a42fca2de32884cd1f1c4b1102ba"
+
+
```

Comment:
It was intentional to simulate what the file would look like if the users had it open in a text editor. If you think that's too goofy, I can get rid of it.